modulus/datapipes/climate/climate.py

```diff
@@ -0,0 +1,829 @@
+
+        # Sanity checks
+        if max(self.channels) >= dataset_shape[1]:
+            raise ValueError(
+                f"Provided channel has indexes greater than the number \
+            of fields {dataset_shape[1]}"
+            )
+
+        if self.num_samples_per_year > data_samples_per_year:
+            raise ValueError(
+                f"num_samples_per_year ({self.num_samples_per_year}) > number of \
+                samples available ({data_samples_per_year})!"
+            )
+
+        self._load_statistics()
+
+        self.logger.info(f"Number of samples/year: {self.num_samples_per_year}")
+        self.logger.info(f"Number of channels available: {dataset_shape[1]}")
+
+    def _load_statistics(self) -> None:
+        """Loads climate statistics from pre-computed numpy files
+
+        The statistic files should be of name global_means.npy and global_std.npy with
+        a shape of [1, C, 1, 1] located in the stat_dir.
+
+        Raises
+        ------
+        IOError
+            If statistics files are not found
+        AssertionError
+            If loaded numpy arrays are not of correct size
+        """
+        # If no stats files we just skip loading the stats
+        if self.stats_files is None:
+            self.mu = None
+            self.std = None
+            return
+        # load normalisation values
+        if set(self.stats_files) == {"mean", "std"}:  # use mean and std files
+            mean_stat_file = self.stats_files["mean"]
+            std_stat_file = self.stats_files["std"]
+
+            if not mean_stat_file.exists():
+                raise IOError(f"Mean statistics file {mean_stat_file} not found")
+            if not std_stat_file.exists():
+                raise IOError(f"Std statistics file {std_stat_file} not found")
+
+            # has shape [1, C, 1, 1]
+            self.mu = np.load(str(mean_stat_file))[:, self.channels]
+            # has shape [1, C, 1, 1]
+            self.sd = np.load(str(std_stat_file))[:, self.channels]
+        elif set(self.stats_files) == {
+            "norm",
+        }:  # use dict formatted file with named variables
+            norm_stat_file = self.stats_files["norm"]
+            if not norm_stat_file.exists():
+                raise IOError(f"Statistics file {norm_stat_file} not found")
+
+            norm = np.load(str(norm_stat_file), allow_pickle=True).item()
+            mu = np.array([norm[var]["mean"] for var in self.variables])
+            self.mu = mu.reshape((1, len(mu), 1, 1))
+            sd = np.array([norm[var]["std"] for var in self.variables])
+            self.sd = sd.reshape((1, len(sd), 1, 1))
+        else:
+            raise ValueError(("Invalid statistics file specification"))
+
+        if not self.mu.shape == self.sd.shape == (1, len(self.channels), 1, 1):
+            raise ValueError("Error, normalisation arrays have wrong shape")
+
+
+class ClimateDatapipe(Datapipe):
```

Comment:
I have a hard time accepting the term 'ClimateDatapipe' when most of the inputs here are not climate related. 'ClimateDatapipe' also seems like a very generic term but this class is quite restrictive.

Reply:
I'm perfectly fine with changing the name, it does seem a bit awkward. I made it up simply by taking the existing `ClimateHDF5Datapipe` and dropping the `HDF5` since it was no longer HDF5-specific.